keymanapp/keyman
# fix(android): Mirror arrow icons for RTL layouts 🍒

Because it will take a while for #5706 to resolve (checking in Arabic strings), this is a 🍒 pick of #5717 which fixes Android arrow icons to mirror for RTL layouts.

## User Testing
We won't have an Arabic local available for testing the Keyman app, so use a [pseudolocale](https://developer.android.com/guide/topics/resources/pseudolocales) AR(XB). This will display English strings (RTL) and allow us to test the RTL layouts

<details><summary>Pseudolocale setup</summary>

1. Start an Android 10 emulator (but don't launch the Keyman app yet)
2. From the device's "Settings" --> About Phone --> Build number --> [enable developer settings](https://developer.android.com/studio/debug/dev-options). This usually involves clicking on the Android Build number multiple times.
3. After developer settings are enabled, reboot the Android emulator
4. When the emulator is back, go to device's "Settings" --> Languages --> cibarA (XB)
![Screenshot_1632186847](https://user-images.githubusercontent.com/7358010/134098370-184a2da3-c13d-4f69-9103-81ae0b59687d.png)
    * If you can't find the (XB) pseudolocale, troubleshoot with the pseudolocale link at the top of this setup summary
</details>

* **TEST_RTL_ARROWS** - Tests that  arrows are correct for RTL
1. With the device in the AR(XB) locale, start the PR build of Keyman for Android
2. Verify the app shows English strings in an RTL layout
![Screenshot_1632187618](https://user-images.githubusercontent.com/7358010/134098490-1e449810-954d-4327-9502-4a98a1b49a02.png)
3. In the app, go to "Settings"
4. On the "Installed languages" selection, verify the forward arrow points left. (Screenshot shows 2 installed languages, but you'll only have 1 off a fresh install)
![Screenshot_1632186959](https://user-images.githubusercontent.com/7358010/134098590-1d72fd3b-f7a7-4e1b-99ac-9fa7c922fd7a.png)
5. Click on "Installed languages" --> English
6. Verify the Forward arrows point left, and the back arrow points right
![Screenshot_1632187808](https://user-images.githubusercontent.com/7358010/134098640-7b288e98-7d9d-4e26-98ee-0458622357cf.png)
7. Back out of Settings
8. From the top menu, select "Text Size" menu
9. Click on the forward arrow (pointing left)
10. Verify the text size increases
11. Click on the back arrow (pointing right)
12. Verify the text size decreases
![Screenshot_1632187865](https://user-images.githubusercontent.com/7358010/134098753-423f9484-43ac-4a4f-8c80-6a64fe097df8.png)
